packages/uploadthing/src/client.ts

```diff
@@ -327,33 +327,45 @@ async function uploadPresignedPost(
   file: File,
   presigned: PSPResponse,
   opts: {
+    reportEventToUT: ReturnType<typeof createUTReporter>;
     onUploadProgress?: UploadFilesOptions<any, any>["onUploadProgress"];
   },
 ) {
   const formData = new FormData();
   Object.entries(presigned.fields).forEach(([k, v]) => formData.append(k, v));
   formData.append("file", file); // File data **MUST GO LAST**
 
-  const response = await new Promise<XMLHttpRequest>((resolve, reject) => {
-    const xhr = new XMLHttpRequest();
-    xhr.open("POST", presigned.url);
-    xhr.setRequestHeader("Accept", "application/xml");
-    xhr.upload.onprogress = (p) => {
-      opts.onUploadProgress?.({
-        file: file.name,
-        progress: (p.loaded / p.total) * 100,
-      });
-    };
-    xhr.onload = (e) => resolve(e.target as XMLHttpRequest);
-    xhr.onerror = (e) => reject(e);
-    xhr.send(formData);
-  });
+  let response: XMLHttpRequest;
```

Comment:
Any reason why you need let + try-catch instead of `.catch()`?

```
const response: XMLHttpRequest = await new Promise<XMLHttpRequest>(
    (resolve, reject) => {
      const xhr = new XMLHttpRequest();
      xhr.open("POST", presigned.url);
      xhr.setRequestHeader("Accept", "application/xml");
      xhr.upload.onprogress = (p) => {
        opts.onUploadProgress?.({
          file: file.name,
          progress: (p.loaded / p.total) * 100,
        });
      };
      xhr.onload = (e) => resolve(e.target as XMLHttpRequest);
      xhr.onerror = (e) => reject(e);
      xhr.send(formData);
    },
  ).catch(async (error) => {
    await opts.reportEventToUT("failure", {
      fileKey: presigned.key,
      uploadId: null,
      fileName: file.name,
      s3Error: (error as Error).toString(),
    });
    throw "unreachable"; // failure event will throw for us
  });
```

Reply:
Probably not - getting resrutten in effect anyway so didnt worry ablut styling too much 😅